### ISSUE TITLE
JOSS review - installation & general structure

### ISSUE THREAD

**Issue by PeerHerholz** [CyclotronResearchCentre/bidsme#4]
Ahoi hoi @nbeliy & co-authors,

this is in reference to the review I'm currently conducting for `JOSS` [here](https://github.com/openjournals/joss-reviews/issues/5575) .

First of all, thank you very much for your dedicated work on this tool and your respective submission to `JOSS`.
Based on the [review process and included checkboxes](https://github.com/openjournals/joss-reviews/issues/5575#issuecomment-1627300948), I would like to suggest a few things.

When I tried to install `bidsme` following the provided [installation guide](https://gitlab.uliege.be/CyclotronResearchCentre/Public/bidstools/bidsme/bidsme_tutorial/-/tree/master/Installation), I ran into an error saying:

`ModuleNotFoundError: No module named 'ruamel'`

and thus, I could not evaluate the other tutorials and performance. It might also be nice to add your package to [pypi](https://pypi.org/).

Furthermore, it might be beneficial to have everything, ie the software, documentation and examples, in the same `github` repository and split between `github` and `gitlab` as this potentially complicates things for interested users (ie finding information, etc.).
Related to that: the [README file](https://github.com/CyclotronResearchCentre/bidsme/blob/dev/README.md) could be enhanced by adding more information, e.g. regarding the installation, etc. .

I'm very sorry if I missed something obvious, misunderstood something or assumed something wrong. Please let me know if that is the case, as well as if you have any questions. Thanks again for creating this important software tool.

Cheers, Peer

**Comment by nbeliy** [comment]
Hi @PeerHerholz ,

- Can you tell how you tried to install `bidsme` exactly? I crosschecked, `ruamel.yaml` is both in `setup.py` and `requirements.txt`, so it should be installed automatically by pip. Maybe I missed a parameter sonwhere...
- I used to have README to contain all information, but it was too bloated and difficult to find anything in it. I'll check if I can add more revelant links to it.
- `github` and `gitlab` -- will fix it

**Comment by PeerHerholz** [comment]
Hi @nbeliy,

thx for the reply and information/explanations!

> Can you tell how you tried to install bidsme exactly? I crosschecked, ruamel.yaml is both in setup.py and requirements.txt, so it should be installed automatically by pip. Maybe I missed a parameter sonwhere...

I followed the provided [installation guide](https://gitlab.uliege.be/CyclotronResearchCentre/Public/bidstools/bidsme/bidsme_tutorial/-/tree/master/Installation).

> I used to have README to contain all information, but it was too bloated and difficult to find anything in it. I'll check if I can add more revelant links to it.

Totally understandable, thanks! 

> github and gitlab -- will fix it

Awesome, thx!

Cheers, Peer

**Comment by nbeliy** [comment]
Hi Peer,

> I followed the provided [installation guide](https://gitlab.uliege.be/CyclotronResearchCentre/Public/bidstools/bidsme/bidsme_tutorial/-/tree/master/Installation). 

That worries me -- I just tried in new environment, and ruamel was installed:
 > Successfully built bidsme
Installing collected packages: pytz, tzdata, ruamel.yaml.clib, numpy, humanfriendly, ruamel.yaml, pandas, coloredlogs, bidsme
Successfully installed bidsme-1.4.0.post4 coloredlogs-15.0.1 humanfriendly-10.0 numpy-1.24.4 pandas-2.0.3 pytz-2023.3 ruamel.yaml-0.17.32 ruamel.yaml.clib-0.2.7 tzdata-2023.3

Can you send me the output of the `pip install` command in tutorial? Maybe a jupyther issue...

**Comment by PeerHerholz** [comment]
Hi @nbeliy,

the `pip install` works fine. I get the error when I run `import bidsme`. 
Sorry, I should've been more precise.

**Comment by nbeliy** [comment]
> the pip install works fine. I get the error when I run import bidsme.

Yes, I understood. The `ruamel` module is supposed to  be installed by pip. Can you confirm that you run bidsme in the same environment where you did `pip install`?

Also, it may be version problem -- if in the later ruamel they renamed the module. Can you give me the python version and ruamel version (you can see it using `pip show  ruamel.yaml`)

**Comment by nbeliy** [comment]
Hi, I added the installation section into README and fixed the links, they points to github now.

Will check the community guides later.

Is the situation with ruamel got clarified, or the issue persists?

**Comment by PeerHerholz** [comment]
Hi @nbeliy,

sorry for the late reply.

>  Hi, I added the installation section into README and fixed the links, they points to github now.

> Will check the community guides later.

Coolio!

> Yes, I understood. The ruamel module is supposed to be installed by pip. Can you confirm that you run bidsme in the same environment where you did pip install?

Yes, it's the same environment.

> Also, it may be version problem -- if in the later ruamel they renamed the module. Can you give me the python version and ruamel version (you can see it using pip show  ruamel.yaml)

I tested it on `python 3.6` and the `ruamel` version I get is `0.15.71`.

**Comment by nbeliy** [comment]
> I tested it on python 3.6 and the ruamel version I get is 0.15.71

Weird, I tried into a new environment, and has no problem with yaml. Out of desperation, can you try following commands, and send me output?
```bash
python3.6 -m venv bidsme-test
source bidsme-test/bin/activate
python3 -m pip install git+https://github.com/CyclotronResearchCentre/bidsme.git
bidsme
```

**Comment by PeerHerholz** [comment]
Hi @nbeliy,

sorry for the late reply.
Creating a new environment resolves the issue and the `CLI` and `python` usage options work. I'm wondering what happened in the environment I tried before. Maybe it's worthwhile adding a small note concerning environments (e.g. "It is recommended to create and utilize a dedicated python environment.")?

**Comment by nbeliy** [comment]
Hi @PeerHerholz, sory for the long silence.

If I understand, the installation issue has been fixed on your side. In the readme, there a suggestion to use virtual envirement:
> It is recommended to use virtual environment when installing bidsme (more info [here](https://github.com/CyclotronResearchCentre/bidsme_tutorial#using-virtual-environments-and-kernels) and [here](https://docs.python.org/3/library/venv.html)).

Is it enough, or should I stress it more and with more instructions?

**Comment by PeerHerholz** [comment]
Hi @nbeliy,

no worries at all! I think it's good. A few thoughts I had: maybe rephrase `UNIX` to `venv` and outline that users only need the `ipykernel` step if they plan to use the respective `bidsme environment` in a `jupyter notebook`.

Sorry if this was already addressed somewhere but do you plan to add documentation in form of a webpage?

**Comment by nbeliy** [comment]
Hi @PeerHerholz,

I uploaded CONTRIBUTING, also added changes that you requested into INSTALLATION.

> Sorry if this was already addressed somewhere but do you plan to add documentation in form of a webpage?

Unfortunately no such plans exists -- it's one man project, and I don't really have time to create full web documentation(

**Comment by PeerHerholz** [comment]
Hi @nbeliy,

thanks implementing the changes and for the information! In that case, could you maybe add a little overview section to the `README.md`, ie linking to all respective parts (contributing, etc.)? Sorry for asking all those add-ons, I just think it might be easier for users to navigate.

**Comment by nbeliy** [comment]
Hi @PeerHerholz ,

Is there something in this issue still unadressed, and if so, can you close it?

Thanks,
Nikita

**Comment by PeerHerholz** [comment]
Hi @nbeliy,

as mentioned in the prior comment, it would be nice to have a little overview with respective links in the `README.md` to make it easier for users to navigate.

Cheers, Peer

**Comment by nbeliy** [comment]
Can you check if new readme is better now?

**Comment by PeerHerholz** [comment]
Hi @nbeliy,

yes, thanks for implementing the changes!
